src/components/GraphFilter/GraphFind.tsx

```diff
@@ -100,10 +166,25 @@ export class GraphFind extends React.PureComponent<GraphFindProps> {
     this.handleFind();
   };
 
-  private handleFindSubmit = event => {
-    event.preventDefault();
-    this.findValue = this.findInputValue;
-    this.handleFind();
+  private handleFilter = () => {
+    if (!this.props.cyData) {
+      console.debug('Skip Filter: cy not set.');
+      return;
+    }
+    const cy = this.props.cyData.cyRef;
+    const selector = this.parseFindValue(this.filterValue);
+    cy.startBatch();
+    // restore old filter-hits
+    if (this.filteredElements) {
+      this.filteredElements.restore();
+      this.filteredElements = undefined;
+      runLayout(cy, this.props.layout);
+    }
+    if (selector) {
+      // remove new filter-hits
+      this.filteredElements = cy.remove(selector);
```

Comment:
I'm not sure.  The problem is that after a refresh an element that was removed may no longer be validly removed.  For example, say the filter is 'http > 50'.  Initially the rate may be 55 and it is removed.  After a refresh it may be 45 and so we need it to come back.  I'm not sure I completely understood your suggestion but I can try playing with it.  I used the remove() and restore() because it seemed to be a built-in way to do this sort of thing.

Reply:
@josejulio I played with it but ended up with basically the same code, just replacing the remove/restore with the style suggestions.  It works but I'm not sure the benefit?  If you have code you want to share that does things in a better way please let me know.

Reply:
> 'm not sure. The problem is that after a refresh an element that was removed may no longer be validly removed. For example, say the filter is 'http > 50'.

Ok, i see the problem. In that situation there should be no difference between remove and display:none